src/material-experimental/mdc-progress-spinner/progress-spinner.html

```diff
@@ -0,0 +1,47 @@
+<div class="mdc-circular-progress" role="progressbar">
+  <div class="mdc-circular-progress__determinate-container">
+    <svg [attr.viewBox]="_viewBox" class="mdc-circular-progress__determinate-circle-graphic"
+         xmlns="http://www.w3.org/2000/svg">
+      <circle [attr.r]="_circleRadius"
+              [style.stroke-dasharray.px]="_strokeCircumference"
+              [style.stroke-dashoffset.px]="_strokeDashOffset"
+              [style.stroke-width.%]="_circleStrokeWidth"
+              class="mdc-circular-progress__determinate-circle"
+              cx="50%" cy="50%"/>
+    </svg>
+  </div>
+  <div class="mdc-circular-progress__indeterminate-container">
+    <div class="mdc-circular-progress__spinner-layer">
+      <div class="mdc-circular-progress__circle-clipper mdc-circular-progress__circle-left">
+        <svg [attr.viewBox]="_viewBox" class="mdc-circular-progress__indeterminate-circle-graphic"
+             xmlns="http://www.w3.org/2000/svg">
+          <circle [attr.r]="_circleRadius"
+                  [style.stroke-dasharray.px]="_strokeCircumference"
+                  [style.stroke-dashoffset.px]="_strokeCircumference/2"
+                  [style.stroke-width.%]="_circleStrokeWidth"
+                  cx="50%" cy="50%"/>
+        </svg>
+      </div>
+      <div class="mdc-circular-progress__gap-patch">
+        <svg [attr.viewBox]="_viewBox" class="mdc-circular-progress__indeterminate-circle-graphic"
+             xmlns="http://www.w3.org/2000/svg">
+          <circle [attr.r]="_circleRadius"
+                  [style.stroke-dasharray.px]="_strokeCircumference"
+                  [style.stroke-dashoffset.px]="_strokeCircumference/2"
+                  [style.stroke-width.%]="_circleStrokeWidth"
+                  cx="50%" cy="50%"/>
+        </svg>
+      </div>
+      <div class="mdc-circular-progress__circle-clipper mdc-circular-progress__circle-right">
+        <svg [attr.viewBox]="_viewBox" class="mdc-circular-progress__indeterminate-circle-graphic"
```

Comment:
It looks like we're repeating the same `svg` markup in 3 different places. You can use `ng-template` to dedupe it:

```
<ng-template #circle>
  <svg><circle/></svg>
</ng-template>

<ng-container [ngTemplateOutlet]="circle"></ng-container>
<ng-container [ngTemplateOutlet]="circle"></ng-container>
```

Reply:
I remember @jelbourn telling me recently to never use ng-templates (obviously an exaggeration, I mean to avoid unnecessary ones because they have performance cost)

Reply:
My thing was about putting `ng-content` _inside_ `ng-template`